code/src/LeafletProvider/DrawingTools/AbstractDrawShape.ts

```diff
@@ -0,0 +1,92 @@
+                OSFramework.Helper.Constants
+                    .shapeChangedEvent as OSFramework.Event.Shape.ShapeEventType,
+                () => {
+                    this.drawingTools.drawingToolsEvents.trigger(
+                        // EventType
+                        OSFramework.Event.DrawingTools.DrawingToolsEventType
+                            .ProviderEvent,
+                        // EventName
+                        this.completedToolEventName,
+                        // The extra parameters, uniqueId and isNewElement set to false indicating that the element is not new
+                        { uniqueId: _shape.uniqueId, isNewElement: false }
+                    );
+                }
+            );
+        }
+
+        /** Create the new shape element based on the configurations (already contains the locations, the bounds or the center and radius depending on the type of the new shape) */
+        protected createShapeElement(
+            uniqueId: string,
+            type: OSFramework.Enum.ShapeType,
+            // eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types, @typescript-eslint/no-explicit-any
+            configs: any
+        ): OSFramework.Shape.IShape {
+            const _shape = Shape.ShapeFactory.MakeShape(
+                this.map,
+                uniqueId,
+                type,
+                configs
+            );
+
+            // Add the onChange event to the new element
+            this._setOnChangeEvent(_shape);
+            // Add the new element to the map
+            this.map.addShape(_shape);
+            return _shape;
+        }
+
+        public build(): void {
+            super.build();
+
+            this.options = this.getProviderConfig();
+        }
+
+        // eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types, @typescript-eslint/no-explicit-any, @typescript-eslint/no-unused-vars
+        public changeProperty(propertyName: string, value: any): void {
+            const propValue = OSFramework.Enum.OS_Config_Shape[propertyName];
+            super.changeProperty(propertyName, value);
+            if (this.drawingTools.isReady) {
+                switch (propValue) {
+                    case OSFramework.Enum.OS_Config_Shape.strokeOpacity:
+                        this.options = { shapeOptions: { opacity: value } };
+                        return;
+                    case OSFramework.Enum.OS_Config_Shape.strokeColor:
+                        this.options = { shapeOptions: { color: value } };
+                        return;
+                    case OSFramework.Enum.OS_Config_Shape.strokeWeight:
+                        this.options = { shapeOptions: { weight: value } };
+                        return;
+                    // If the following configurations are not included on the configs of the tool, the AbstractTool will make sure to throw an error
```

Comment:
This comment should be moved to the start or end of the switch. Doesn't make sense to have this here in the middle